### PR TITLE
DMP-2089 - indexes v1, v2 and v3

### PR DESCRIPTION
**Pull request by cmagowan-hmcts**
https://tools.hmcts.net/jira/browse/DMP-2089

### CODE REVIEW

File: src/main/resources/db/migration/postgres/V1_271__indexes_v1_v2_v3.sql [hunk 1]
```diff
@@ -0,0 +1,182 @@
+SET SEARCH_PATH TO darts;
+
+CREATE INDEX ann_cur_own_fk     ON ANNOTATION(current_owner);
+CREATE INDEX ann_cre_by_fk      ON ANNOTATION(created_by);
+CREATE INDEX ann_lst_mod_by_fk  ON ANNOTATION(last_modified_by);
+
+CREATE INDEX ado_ann_fk         ON ANNOTATION_DOCUMENT(ann_id);
+
+CREATE INDEX aud_cas_fk         ON AUDIT(cas_id);
```
Comment (mestebanez) [inline src/main/resources/db/migration/postgres/V1_271__indexes_v1_v2_v3.sql:9]: I'm genuinely curious. Is there any advantage to indexing numbers?

File: src/main/resources/db/migration/postgres/V1_271__indexes_v1_v2_v3.sql [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+CREATE INDEX esa_cre_by_fk      ON EXTERNAL_SERVICE_AUTH_TOKEN(created_by);
+CREATE INDEX esa_lst_mod_by_fk  ON EXTERNAL_SERVICE_AUTH_TOKEN(last_modified_by);
+
+CREATE INDEX hea_cas_fk         ON HEARING(cas_id);
+CREATE INDEX hea_ctr_fk         ON HEARING(ctr_id);
+CREATE INDEX hea_cre_by_fk      ON HEARING(created_by);
+CREATE INDEX hea_lst_mod_by_fk  ON HEARING(last_modified_by);
+
+--CREATE INDEX haa_hea_fk         ON HEARING_ANNOTATION_AE(hea_id);
```
Comment (mestebanez) [inline src/main/resources/db/migration/postgres/V1_271__indexes_v1_v2_v3.sql:75]: Do we need these commented out indexes?

File: src/main/resources/db/migration/postgres/V1_271__indexes_v1_v2_v3.sql [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+CREATE INDEX not_lst_mod_fk     ON NOTIFICATION(last_modified_by);
+
+CREATE INDEX prn_cas_fk         ON PROSECUTOR(cas_id);
+CREATE INDEX prn_cre_by_fk      ON PROSECUTOR(created_by);
+CREATE INDEX prn_lst_mod_by_fk  ON PROSECUTOR(last_modified_by);
+
+CREATE INDEX rep_cre_by_fk      ON REPORT(created_by);
+CREATE INDEX rep_lst_mod_by_fk  ON REPORT(last_modified_by);
+
+--CREATE INDEX tra_cas_fk         ON TRANSCRIPTION(cas_id);
```
Comment (mestebanez) [inline src/main/resources/db/migration/postgres/V1_271__indexes_v1_v2_v3.sql:114]: Same?

File: src/main/resources/db/migration/postgres/V1_271__indexes_v1_v2_v3.sql [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+--v19 security
+CREATE INDEX grp_rol_fk         ON SECURITY_GROUP(rol_id);
+CREATE INDEX grp_cre_by_fk      ON SECURITY_GROUP(created_by);
+CREATE INDEX grp_lst_mod_by_fk  ON SECURITY_GROUP(last_modified_by);
+
+--CREATE INDEX grc_grp_fk         ON SECURITY_GROUP_COURTHOUSE_AE(grp_id);
+CREATE INDEX grc_cth_fk         ON SECURITY_GROUP_COURTHOUSE_AE(cth_id);
+
+CREATE INDEX gua_grp_fk         ON SECURITY_GROUP_USER_ACCOUNT_AE(grp_id);
+--CREATE INDEX gua_usr_fk         ON SECURITY_GROUP_USER_ACCOUNT_AE(usr_id);
```
Comment (mestebanez) [inline src/main/resources/db/migration/postgres/V1_271__indexes_v1_v2_v3.sql:151]: Same?

File: src/main/resources/db/migration/postgres/V1_271__indexes_v1_v2_v3.sql (hunk 3 above)
Comment (cmagowan-hmcts) [reply src/main/resources/db/migration/postgres/V1_271__indexes_v1_v2_v3.sql]: removed

File: src/main/resources/db/migration/postgres/V1_271__indexes_v1_v2_v3.sql (hunk 4 above)
Comment (cmagowan-hmcts) [reply src/main/resources/db/migration/postgres/V1_271__indexes_v1_v2_v3.sql]: removed

**Review by hemantasharma1129 — APPROVED** [review]

**Pull request merged**